src/NUnitFramework/framework/Internal/TypeNameDifferenceResolverUtils.cs

```diff
@@ -0,0 +1,163 @@
+                genericParams.ForEach(x => shortenedGenericParams.Add(FullyShortenTypeName(x)));
+
+                return ReconstructGenericTypeName(genericType, shortenedGenericParams);
+            }
+            else
+            {
+                return GenericType.Name;
+            }
+        }
+
+        /// <summary>
+        /// Shorten the given <see cref="Type"/> names by only including the relevant differing namespaces/types, if they differ.
+        /// </summary>
+        /// <param name="expectedType">The expected <see cref="Type"/>.</param>
+        /// <param name="actuallType">The actual <see cref="Type"/>.</param>
+        /// <param name="expectedTypeShortened">The shortened expected <see cref="Type"/> name.</param>
+        /// <param name="actualTypeShortened">The shortened actual <see cref="Type"/> name.</param>
+        public void ShortenTypeNames(Type expectedType, Type actuallType, out string expectedTypeShortened, out string actualTypeShortened)
+        {
+            string[] expectedOriginalType = expectedType.FullName.Split('.');
+            string[] actualOriginalType = actuallType.FullName.Split('.');
+
+            bool diffDetected = false;
+            int actualStart = 0, expectStart = 0;
+            for (int expectLen = expectedOriginalType.Length - 1, actualLen = actualOriginalType.Length - 1;
+                expectLen >= 0 && actualLen >= 0;
+                expectLen--, actualLen--)
+            {
+                if (expectedOriginalType[expectLen] != actualOriginalType[actualLen])
+                {
+                    actualStart = actualLen;
+                    expectStart = expectLen;
+                    diffDetected = true;
+                    break;
+                }
+            }
+
+            if (diffDetected)
+            {
+                expectedTypeShortened = String.Join(".", expectedOriginalType, expectStart, expectedOriginalType.Length - expectStart);
+                actualTypeShortened = String.Join(".", actualOriginalType, actualStart, actualOriginalType.Length - actualStart);
+            }
+            else
+            {
+                expectedTypeShortened = expectedOriginalType[expectedOriginalType.Length - 1];
+                actualTypeShortened = actualOriginalType[actualOriginalType.Length - 1];
+            }
+
+        }
+
+        /// <summary>
+        /// Returns whether or not the <see cref="Type"/> is generic.
+        /// </summary>
+        public bool IsTypeGeneric(Type type)
+        {
+            Guard.ArgumentNotNull(type, nameof(type));
+
+            return type.GetGenericArguments().Length > 0;
+        }
+
+        /// <summary>
+        /// Get the generic parameters of a given <see cref="Type"/>.
+        /// </summary>
+        public List<Type> GetGenericParams(Type type)
+        {
+            Guard.ArgumentNotNull(type, nameof(type));
+
+            return new List<Type>(type.GetGenericArguments());
```

Comment:
I'm curious why we don't use `type.GetGenericArguments()` instead of maintaining the helper method? Also arguments and parameters are not the same thing, though the reflection API is already abusing the distinction...

Reply:
I think I intended to refactor that away after I refactored my functionality to use `Type`s instead of relying on the string representation of a `Type`. I think you're right, we don't need a new method for that, so as long as I put `using NUnit.Compatability;` at the top of the file!